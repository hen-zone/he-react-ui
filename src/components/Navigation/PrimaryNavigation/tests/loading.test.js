import createTestContext from 'react-cosmos-test/enzyme'
import fixture from '../fixtures/Loading.fixture'

const { mount, getWrapper } = createTestContext({ fixture })

beforeEach(mount)

test('<PrimaryNavigation /> rendered correctly with loading fixture', () => {
  let wrapper = getWrapper()

  expect(wrapper).toMatchSnapshot()
})
