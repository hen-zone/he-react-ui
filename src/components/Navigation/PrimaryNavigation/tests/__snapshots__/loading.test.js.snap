// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryNavigation /> rendered correctly with loading fixture 1`] = `
<OnClickOutside(PrimaryNavigation)
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  excludeScrollbar={false}
  items={Array []}
  loading={true}
  locations={Array []}
  logoutRoute="/logout"
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  preventDefault={false}
  stopPropagation={false}
>
  <PrimaryNavigation
    bottomKeys={Array []}
    disableOnClickOutside={[Function]}
    enableOnClickOutside={[Function]}
    eventTypes={
      Array [
        "mousedown",
        "touchstart",
      ]
    }
    items={Array []}
    loading={true}
    locations={Array []}
    logo={
      Object {
        "icon": <Icon
          className="logo"
          inverted={true}
          name="HealthEngine"
        />,
        "route": "/",
      }
    }
    logoutRoute="/logout"
    outsideClickIgnoreClass="ignore-react-onclickoutside"
    preventDefault={false}
    stopPropagation={false}
  >
    <div
      className="outer"
    >
      <div
        className="nav"
      >
        <div
          className="buckets"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="logoBucket"
            onClick={[Function]}
            to="/"
          >
            <Route
              path="/"
            >
              <Link
                aria-current="true"
                className="logoBucket active"
                onClick={[Function]}
                replace={false}
                style={Object {}}
                to="/"
              >
                <a
                  aria-current="true"
                  className="logoBucket active"
                  href="/"
                  onClick={[Function]}
                  style={Object {}}
                >
                  <div
                    className="logo"
                  >
                    <Icon
                      className="logo"
                      inverted={true}
                      name="HealthEngine"
                    >
                      <HealthEngine
                        className="icon logo"
                        inverted={true}
                      >
                        <svg
                          className="icon logo"
                          version="1.1"
                          viewBox="0 0 48 38"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            transform="translate(0.000000, 0.000000)"
                          >
                            <path
                              d="M12.894883,32.7452837 C13.1630685,35.6739197 15.6192085,37.9154191 18.5600981,37.9154191 C21.5009878,37.9154191 23.9571278,35.6739197 24.2253133,32.7452837 L24.2253133,25.1600585 L12.894883,25.1600585 L12.894883,32.7452837 L12.894883,32.7452837 Z"
                              fill="#5BC0C9"
                            />
                            <path
                              d="M24.2253133,6.2159583 L24.2253133,6.2159583 C24.4219874,4.06823902 23.387262,1.99397789 21.553185,0.8592795 C19.719108,-0.275418886 17.4010882,-0.275418886 15.5670112,0.8592795 C13.7329342,1.99397789 12.6982088,4.06823902 12.894883,6.2159583 L12.894883,6.2159583 L12.894883,13.8011836 L24.2253133,13.8011836 L24.2253133,6.2159583 Z"
                              fill="#5BC0C9"
                            />
                            <path
                              d="M37.489976,25.1410954 L31.801057,25.1410954 L31.801057,32.7263207 C32.0692425,35.6549566 34.5253825,37.896456 37.4662721,37.896456 C40.4071618,37.896456 42.8633018,35.6549566 43.1314873,32.7263207 L43.1314873,19.4521764 C43.1315957,22.5756585 40.6133496,25.1150664 37.489976,25.1410954 L37.489976,25.1410954 Z"
                              fill="#C6E400"
                            />
                            <path
                              d="M37.489976,0.546002402 C34.3480728,0.546002402 31.801057,3.09301818 31.801057,6.23492137 L31.801057,13.8201467 L37.489976,13.8201467 C40.6318792,13.8201467 43.1788949,16.3671624 43.1788949,19.5090656 L43.1788949,6.23492137 C43.1788949,3.09301818 40.6318792,0.546002402 37.489976,0.546002402 L37.489976,0.546002402 Z"
                              fill="#C6E400"
                            />
                            <path
                              d="M5.17013534,14.0434783 C2.24149938,14.3116637 8.42129176e-16,16.7678037 0,19.7086934 C-8.42129176e-16,22.6495831 2.24149938,25.1057231 5.17013534,25.3739085 L12.7553606,25.3739085 L12.7553606,14.0434783 L5.17013534,14.0434783 L5.17013534,14.0434783 Z"
                              fill="#FFFFFF"
                            />
                            <path
                              d="m 24.225313,13.820147 h 7.604189 v 11.33043 h -7.604189 z"
                              fill="#FFFFFF"
                            />
                            <path
                              d="m 12.894883,13.820147 h 11.33043 v 11.33043 h -11.33043 z"
                              fill="#008EC8"
                            />
                            <path
                              d="M37.7640488,13.8201467 L31.801057,13.8201467 L31.801057,25.1410954 L37.7640488,25.1410954 C40.8113797,24.8492731 43.1314873,22.4024989 43.1314873,19.480621 C43.1314873,16.5587431 40.8113797,14.111969 37.7640488,13.8201467 L37.7640488,13.8201467 Z"
                              fill="#48B400"
                            />
                          </g>
                        </svg>
                      </HealthEngine>
                    </Icon>
                  </div>
                </a>
              </Link>
            </Route>
          </NavLink>
          <LoadingStrip
            className="loadingBucket"
          >
            <div
              className="outer loadingBucket"
            >
              <div
                className="inner"
              />
            </div>
          </LoadingStrip>
          <LoadingStrip
            className="loadingBucket"
          >
            <div
              className="outer loadingBucket"
            >
              <div
                className="inner"
              />
            </div>
          </LoadingStrip>
          <LoadingStrip
            className="loadingBucket bottom"
          >
            <div
              className="outer loadingBucket bottom"
            >
              <div
                className="inner"
              />
            </div>
          </LoadingStrip>
        </div>
      </div>
      <div
        className="spacer"
      />
      <div
        className="content"
        onClick={[Function]}
      >
        <withRouter(SubNavigation)
          loading={true}
          logoutRoute="/logout"
        >
          <Route
            render={[Function]}
          >
            <SubNavigation
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "mocked",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "mocked",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              loading={true}
              location={
                Object {
                  "hash": "",
                  "key": "mocked",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              logoutRoute="/logout"
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <div
                className="bar"
              >
                <div
                  className="top"
                >
                  <Heading
                    className="heading"
                    h1={true}
                    h2={false}
                    h3={false}
                    h4={false}
                    nav={false}
                  >
                    <h1
                      className="heading h1 heading"
                    >
                      <LoadingStrip
                        className="loading"
                      >
                        <div
                          className="outer loading"
                        >
                          <div
                            className="inner"
                          />
                        </div>
                      </LoadingStrip>
                    </h1>
                  </Heading>
                  <div
                    className="controls"
                  >
                    <span
                      className="control"
                    >
                      <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="navLink"
                        key="logout"
                        target="_self"
                        title="Logout"
                        to="/logout"
                      >
                        <Route
                          path="/logout"
                        >
                          <Link
                            aria-current={false}
                            className="navLink"
                            replace={false}
                            target="_self"
                            title="Logout"
                            to="/logout"
                          >
                            <a
                              aria-current={false}
                              className="navLink"
                              href="/logout"
                              onClick={[Function]}
                              target="_self"
                              title="Logout"
                            >
                              <Icon
                                className="icon"
                                name="Logout"
                              >
                                <Logout
                                  className="icon icon"
                                >
                                  <svg
                                    className="icon icon"
                                    version="1.1"
                                    viewBox="0 1 16 16"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M 7.5 0.66601562 C 6.6774686 0.66601562 6 1.3434842 6 2.1660156 L 6 5.1660156 L 7 5.1660156 L 7 2.1660156 C 7 1.884547 7.2185314 1.6660156 7.5 1.6660156 L 14.5 1.6660156 C 14.781469 1.6660156 15 1.884547 15 2.1660156 L 15 14.166016 C 15 14.447484 14.781469 14.666016 14.5 14.666016 L 7.5 14.666016 C 7.2185314 14.666016 7 14.447484 7 14.166016 L 7 11.166016 L 6 11.166016 L 6 14.166016 C 6 14.988547 6.6774686 15.666016 7.5 15.666016 L 14.5 15.666016 C 15.322531 15.666016 16 14.988547 16 14.166016 L 16 2.1660156 C 16 1.3434842 15.322531 0.66601562 14.5 0.66601562 L 7.5 0.66601562 z M 3.1464844 4.8125 L -0.20703125 8.1660156 L 3.1464844 11.519531 L 3.8535156 10.8125 L 1.7070312 8.6660156 L 11.5 8.6660156 L 11.5 7.6660156 L 1.7070312 7.6660156 L 3.8535156 5.5195312 L 3.1464844 4.8125 z "
                                      transform="translate(0,1)"
                                    />
                                  </svg>
                                </Logout>
                              </Icon>
                              Logout
                            </a>
                          </Link>
                        </Route>
                      </NavLink>
                    </span>
                  </div>
                </div>
              </div>
            </SubNavigation>
          </Route>
        </withRouter(SubNavigation)>
        <div
          className="children"
        />
      </div>
    </div>
  </PrimaryNavigation>
</OnClickOutside(PrimaryNavigation)>
`;
